### PR TITLE
Feature/lam 79

### PR DESCRIPTION
**Pull request by mclaurentiu**
Checks are failing because of the bug https://meaningfy.atlassian.net/browse/LAM-147

### CODE REVIEW

File: tests/steps/test_cases_functional.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        from tests.config import LAM_URL
+        scenario_context["base_uri"] = LAM_URL
+
+
+@when(parsers.cfparse('I navigate to the location {pageLocation:String}', extra_types=dict(String=str)))
+def i_navigate_to_the_location_page(scenario_context, browser, pageLocation):
+    browser.get(scenario_context["base_uri"] + pageLocation)
+
+
+@then(parsers.cfparse('the resulting page contains {content:String} in the element with id {field_id:String}',
```
Comment (cosletmihai) [inline tests/steps/test_cases_functional.py:80]: it doesn't seem that there are `feature` steps with this text, consider removing them?

File: tests/steps/test_cases_functional.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+@then(parsers.cfparse('the field with XPath {field_xpath:String} is visible', extra_types=dict(String=str)))
+def step_impl(browser, scenario_context, field_xpath):
+    sleep(1)
+    assert browser.find_element_by_xpath(field_xpath).is_displayed() is True
+
+
+@then(parsers.cfparse('the element with XPath {field_xpath:String} is in the viewport', extra_types=dict(String=str)))
+def step_impl(browser, scenario_context, field_xpath):
+    sleep(1)
```
Comment (cosletmihai) [inline tests/steps/test_cases_functional.py:130]: is `1` second enough for the page to render properly?

File: tests/steps/test_cases_functional.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                      extra_types=dict(String=str)))
+def i_fill_in_the_field_field_id_sparql_with_someendpointhere(scenario_context, browser, control_id, text_value):
+    browser.find_element_by_id(control_id).send_keys(scenario_context[text_value])
+
+
+def is_element_visible_in_viewpoint(driver, element) -> bool:
+    return driver.execute_script("var elem = arguments[0],                 "
+                                 "  box = elem.getBoundingClientRect(),    "
+                                 "  cx = box.left + box.width / 2,         "
+                                 "  cy = box.top + box.height / 2,         "
```
Comment (cosletmihai) [inline tests/steps/test_cases_functional.py:188]: hack alert? 😄

File: tests/steps/test_cases_functional.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+def step_impl(browser, scenario_context, field_xpath):
+    sleep(1)
+    assert browser.find_element_by_xpath(field_xpath).is_displayed() is True
+
+
+@then(parsers.cfparse('the element with XPath {field_xpath:String} is in the viewport', extra_types=dict(String=str)))
+def step_impl(browser, scenario_context, field_xpath):
+    sleep(1)
+    element = browser.find_element_by_xpath(field_xpath)
+    assert is_element_visible_in_viewpoint(browser, element) is True
```
Comment (cosletmihai) [inline tests/steps/test_cases_functional.py:132]: [it is discouraged](https://stackoverflow.com/questions/4591125/is-it-safe-to-replace-with-is-to-compare-boolean-values#answer-4591139) to use the `is` comparator for a boolean value, and neither the `!=` and `==` is required for this assertion. The recommended approach is:
```python
assert is_element_visible_in_viewpoint(browser, element)
```
or if you need to check if the value is `False`:
```python
assert not is_element_visible_in_viewpoint(browser, element)
```

**Review by cosletmihai — APPROVED** [review]

File: tests/steps/test_cases_functional.py (hunk 2 above)
Comment (mclaurentiu) [reply tests/steps/test_cases_functional.py:130]: It is about the animation to stop.

File: tests/steps/test_cases_functional.py (hunk 3 above)
Comment (mclaurentiu) [reply tests/steps/test_cases_functional.py:188]: Couldn't find another way.

**Pull request merged**